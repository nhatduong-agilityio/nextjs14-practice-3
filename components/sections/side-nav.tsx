// Components
import { Separator } from '../ui/separator'
import { Sheet } from '../ui/sheet'
import { NavLinks } from './nav-links'

// Features
import { TeamSelection } from '@/features/teams/components/team-selection'

export const SideNav = () => (
  <div className='flex flex-col pt-10 gap-2.5'>
    <section className='flex flex-col gap-[30px] px-5'>
      <TeamSelection />
      <Separator className='px-5' />
    </section>

    <div>
      <Sheet>
        <NavLinks />
      </Sheet>
    </div>
  </div>
)
