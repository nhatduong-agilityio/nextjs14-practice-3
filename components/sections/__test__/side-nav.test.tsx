import React from 'react'
import { render } from '@testing-library/react'
import { SideNav } from '../side-nav'

jest.mock('@/features/teams/components/team-selection', () => ({
  TeamSelection: () => <div>Team Selection Mock</div>,
}))

jest.mock('../nav-links', () => ({
  NavLinks: () => <div>Nav Links Mock</div>,
}))

describe('SideNav', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(<SideNav />)
    expect(asFragment()).toMatchSnapshot()
  })
})
