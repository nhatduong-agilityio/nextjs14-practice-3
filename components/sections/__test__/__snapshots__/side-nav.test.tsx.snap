// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideNav matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="flex flex-col pt-10 gap-2.5"
  >
    <section
      class="flex flex-col gap-[30px] px-5"
    >
      <div>
        Team Selection Mock
      </div>
      <div
        class="h-[1px] w-full shrink-0 bg-separator px-5"
        data-orientation="horizontal"
        role="none"
      />
    </section>
    <div>
      <div>
        Nav Links Mock
      </div>
    </div>
  </div>
</DocumentFragment>
`;
